glue/core/tests/test_data.py

```diff
@@ -1132,3 +1132,56 @@ def test_compute_histogram_dask_mixed():
 
     result = data.compute_histogram([data.id['y'], data.id['x']], range=[[-0.5, 11.25], [-0.5, 11.25]], bins=[2, 3], subset_state=data.id['x'] > 4.5)
     assert_allclose(result, [[0, 1, 0], [0, 2, 2]])
+
+
+def test_compute_histogram_random_subset():
+
+    # Make sure that compute_histogram works with the random_subset option
+
+    with NumpyRNGContext(12345):
+
+        data = Data(x=np.random.uniform(0, 10, 1024**2), y=np.ones(1024 ** 2))
```

Comment:
I don't see a test for 2D image like how Jdaviz would store it. We have a 2D array under one component. Here, you store each dimension as a separate component. I don't think it is equivalent?

Since we do not use histogram viewer directly in Jdaviz, I am trying to emulate this calculation directly over at https://github.com/spacetelescope/jdaviz/pull/2763 but without success. There must be some subtlety I am missing. Are you able to have a look over there and tell me if I went down the wrong rabbit hole?

Cami has a 8GB test image if you need one, if she is willing to share with you.